ghwatch.py

```diff
@@ -0,0 +1,316 @@
+            # Issue: https://api.github.com/repos/bitcoin/bitcoin/issues/20935
+            m = re.match('.*\/([0-9]+)$', rec.subject.url)
+            issue = int(m.group(1))
+            meta = ghmeta.get((rec.repository.full_name, issue))
+            ref_str = str(issue)
+        elif rec.subject.type == 'Commit':
+            # Commit: https://api.github.com/repos/bitcoin/bitcoin/commits/54ce4fac80689621dcbcc76169b2b00b179ee743
+            m = re.match('.*\/([0-9a-f]+)$', rec.subject.url)
+            ref_str = m.group(1)
+            issue = None
+            meta = None
+        else:
+            # Release: https://api.github.com/repos/bitcoin-core/HWI/releases/34442950
+            # RepositoryInvitation: ?
+            if rec.subject.type not in {'Release', 'RepositoryInvitation'}: # Huh
+                print(rec.subject.type, rec.subject.url)
+                assert(False)
+            issue = None
+            meta = None
+
+        label_t = (Theme.ROW, '')
+        state = 'unknown'
+        if meta is not None:
+            label = pick_label(config['label_prio'], rec.repository.full_name, meta['labels'])
+            if label is not None:
+                label_t = (Attr.bg_hex(label['color']) + Attr.fg(0, 0, 0), label['name'])
+
+            state = meta['state']
+            if meta['pr'] is not None and meta['pr']['merged']:
+                state = 'merged'
+
+        pr.print_row([
+            (Theme.DATETIME, rec.updated_at),
+            Theme.REASON_GLYPHS.get(rec.reason, Theme.UNK_REASON),
+            (Theme.ROW, rec.repository.full_name),
+            (Theme.ROW, rec.subject.type),
+            (Theme.REF.get(state, ''), ref_str),
+            label_t,
+            (Theme.ROW, rec.subject.title),
+            ])
+
+        buttons.append(ButtonInfo(
+            x0 = issue_column.x,
+            y0 = row + 1,
+            x1 = issue_column.x + issue_column.width,
+            y1 = row + 2,
+            url = get_html_url(config['ghbase'], rec),
+        ))
+
+        row += 1
+        if row == N:
+            break
+
+    return buttons
+
+buttons = draw()
+Key.start(hide_cursor=True)
+
+t = 0
+try:
+    while True:
+        # auto-refresh periodically
+        if t >= args.refresh_time:
+            buttons = draw()
+            t = 0
+
+        # handle key input
+        k = Key.get()
+        if not k:
+            Key.input_wait(1.0)
+            t += 1
+            continue
+        if k == 'escape':
+            break
+        # print(k, Key.mouse_pos)
+        if k == 'mouse_click':
+            # TODO: highlight button when clicked for a bit of feedback?
+            for b in buttons:
+                if b.x0 <= Key.mouse_pos[0] < b.x1 and b.y0 <= Key.mouse_pos[1] < b.y1:
+                    if b.url is not None:
+                        subprocess.call(config['browser'] + [b.url], stdout = subprocess.PIPE, stderr = subprocess.PIPE)
```

Comment:
I had to switch this to `webbrowser.open(b.url)` (`import webbrowser`) to work seamlessly on macOS.

Reply:
Wait, is that a built-in thing in Python?
That does seem much better than having a custom setting, though I wonder how it "knows", and how to configure it, will look into it!